### ISSUE TITLE
Add ability to register a11y matcher in  jest config once per project

### ISSUE THREAD

**Issue by mohanraj-r** [salesforce/sa11y#11]
### Motivation 
The a11y matcher API has to be registered before its used. Currently this needs to be done in each test module as part of test setup as shown below. 

```js
import { registerA11yMatchers } from '@sa11y/jest';
beforeAll(() => {
    registerA11yMatchers();
});
describe('integration test @sa11y/jest', () => {
    it('should have a11y matchers working', () => {
        expect(expect.toBeAccessible).toBeDefined();
        expect(document).toBeAccessible();
    });
});
```

This could be made less tedious by registering the a11y matchers *once* in the jest config of the consuming project and then all the tests in the project can use the a11y API without any additional setup. 

For example a consuming project should be able to register a11y matchers by extending the config as below in the `jest.config.js` at the project root:
```javascript
const { jestConfig } = require('@sa11y/jest');

module.exports = {
    ...jestConfig,
    // Rest of the Project's Jest config ..
};
```

### Resources
Similar setup can be found in
* [jest-community/jest-extended: Additional Jest matchers 🃏💪](https://github.com/jest-community/jest-extended#setup)
* [salesforce/sfdx-lwc-jest: Run Jest against LWC components in SFDX workspace environment](https://github.com/salesforce/sfdx-lwc-jest#overriding-jest-config)

**Comment by AllanOricil** [comment]
If you came here because you had this error, be aware that the Api has changed from `registerA11yMatchers` to `setup` while initializing these custom matchers


https://www.npmjs.com/package/@sa11y/jest
````
const { setup } = require('@sa11y/jest'); // CommonJS
import { setup } from '@sa11y/jest'; // ES6
// Register the sa11y matcher
setup();
````
